m/configDataPreprocessingSeaexplorer.m

```diff
@@ -0,0 +1,129 @@
+function preprocessing_options = configDataPreprocessingSeaexplorer()
+%CONFIGDATAPREPROCESSINGSEAEXPLORER  Configure Seaexplorer glider data preprocessing.
+%
+%  Syntax:
+%    PREPROCESSING_OPTIONS = CONFIGDATAPREPROCESSINGSEAEXPLORER()
+%
+%  Description:
+%    PREPROCESSING_OPTIONS = CONFIGDATAPREPROCESSINGSEAEXPLORER() should return 
+%    a struct setting the options for Seaexplorer glider data preprocessing 
+%    as needed by the function PREPROCESSGLIDERDATA.
+%
+%  Examples:
+%    preprocessing_options = configDataPreprocessingSeaexplorer()
+%
+%  See also:
+%    PREPROCESSGLIDERDATA
+%
+%  Authors:
+%    Frederic Cyr  <Frederic.Cyr@mio.osupytheas.fr>
+%    Joan Pau Beltran  <joanpau.beltran@socib.cat>
+
+%  Copyright (C) 2013-2015
+%  ICTS SOCIB - Servei d'observacio i prediccio costaner de les Illes Balears
+%  <http://www.socib.es>
+%
+%  This program is free software: you can redistribute it and/or modify
+%  it under the terms of the GNU General Public License as published by
+%  the Free Software Foundation, either version 3 of the License, or
+%  (at your option) any later version.
+%
+%  This program is distributed in the hope that it will be useful,
+%  but WITHOUT ANY WARRANTY; without even the implied warranty of
+%  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+%  GNU General Public License for more details.
+%
+%  You should have received a copy of the GNU General Public License
+%  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+  error(nargchk(0, 0, nargin, 'struct'));
+  
+  preprocessing_options = struct();
+
+  %% Navigation
+  preprocessing_options.time_list(1).time = 'Posixtime_sci';
+  preprocessing_options.position_list(1).longitude = 'NAV_LONGITUDE';
+  preprocessing_options.position_list(1).latitude  = 'NAV_LATITUDE';
+  preprocessing_options.position_list(1).conversion = @nmea2deg;
+  
+  %preprocessing_options.depth_list(1).depth = 'NAV_DEPTH';
+  preprocessing_options.depth_list(1).depth = 'GPCTD_PRESSURE';
+  preprocessing_options.depth_list(2).depth = 'SBD_PRESSURE';
+  %preprocessing_options.depth_list(3).depth = 'NAV_DEPTH';
+
```

Comment:
In that case I would suggest to keep the navigation depth
(ordered list of preferred sources: `NAV_DEPTH`,`Depth`) as `depth`,
and allow the depth from pressure derivation, resulting in the `depth_ctd` variable.
The rationale is that the GPCTD sampling might be disabled in some stage of a deployment while other sensors are left on, and there will be no vertical coordinate for those readings
(or a very poorly estimated one by interpolation of the previous and next
pressure measurements, which could be too far away),
Actually this is a common situation if the CTD is configured to sample only on downcasts.
Having both depth sequences in the final processed data allows the end user
to choose the best vertical reference coordinate for the data he wants to consume.

The `depth_ctd` is derived from pressure and latitude using CSIRO's `sw_dpth`
in [`processGliderData.m`][depth_ctd].
  [depth_ctd]: <https://github.com/socib/glider_toolbox/blob/master/m/processing_tools/processGliderData.m#L631>